activesupport/lib/active_support/message_pack/serializer.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  module MessagePack
+    module Serializer # :nodoc:
+      attr_accessor :message_pack_factory
+
+      def dump(object)
+        @packer_key ||= "message_pack_packer_#{object_id}"
+        packer = (IsolatedExecutionState[@packer_key] ||= message_pack_factory.packer)
```

Comment:
This may be an issue if someone end up calling that `dump` or `load` method recusively, e.g. from their custom `to_msgpack_ext` method.

Ideally we'd use `MessagePack::Factory::Pool` https://github.com/msgpack/msgpack-ruby/#pooling

Reply:
Oh, excellent point!  But if we're going to have a leading type signature, we'll need `Pool#packer` and `Pool#unpacker` from https://github.com/msgpack/msgpack-ruby/commit/a41195c17758ec967c98a35a03824229e39161ca, which hasn't been released yet.  In the mean time, I could add something like `packer = message_pack_factory.packer if !packer.empty?` after the `IsolatedExecutionState` fetch.